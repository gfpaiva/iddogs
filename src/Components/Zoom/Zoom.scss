@import '../../Scss/root';

.zoom {
  @media screen and (max-width: $xs-beakpoint) { width: 95% }
  left: 50%;
  position: fixed;
  top: 50%;
  transform: translate(-50%, -50%);
  z-index: 15;

  &__mask {
    background-color: rgba(0, 0, 0, .8);
    height: 100%;
    left: 0;
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 10;
  }
}
