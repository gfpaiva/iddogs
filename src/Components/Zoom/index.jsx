/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import './Zoom.scss';

const getCategoryURL = (URL) => {
  const urlToArray = URL.split('/');

  urlToArray.pop();

  return urlToArray.join('/');
};

function Zoom({
  image,
  id,
  setZoom,
  history: { push },
  location: { pathname },
}) {
  const exitZoom = () => {
    setZoom(false);
    push(getCategoryURL(pathname));
  };

  const keyPressHandler = (e) => {
    e.preventDefault();

    if (e.key === 'Escape') exitZoom();
  };

  useEffect(() => {
    document.addEventListener('keyup', keyPressHandler);

    return () => {
      document.removeEventListener('keyup', keyPressHandler);
    };
  }, []);

  return (
    <>
      <div className="zoom text-center">
        <img className="full-width animated zoomIn faster" src={image} alt={id} title={id} />
      </div>
      <div
        className="zoom__mask animated fadeIn faster"
        onClick={exitZoom}
      />
    </>
  );
}


Zoom.propTypes = {
  image: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  setZoom: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

export default withRouter(Zoom);
