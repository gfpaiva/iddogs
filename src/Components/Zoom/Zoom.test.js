import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';''

import feed from '../../../__mocks__/getFeed';

import Zoom from './index';

describe('<Zoom />', () => {
  it('should mount properly', () => {
    const setZoom = jest.fn();

    const wrapper = mount(
      <Router>
        <Zoom
          id={0}
          image={feed.list[0]}
          setZoom={setZoom}
        />
      </Router>
    );

    expect(wrapper.find('.zoom')).toMatchSnapshot();
  });

  it('should close on click in overlay', () => {
    const setZoom = jest.fn();

    const wrapper = mount(
      <MemoryRouter initialEntries={[{ pathname: '/feed/husky/0' }]}>
        <Zoom
          id={0}
          image={feed.list[0]}
          setZoom={setZoom}
        />
      </MemoryRouter>
    );

    wrapper.find('.zoom__mask').simulate('click');
    expect(setZoom).toHaveBeenCalled();
  });
});
