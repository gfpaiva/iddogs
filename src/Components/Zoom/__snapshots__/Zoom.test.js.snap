// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Zoom /> should mount properly 1`] = `
<div
  className="zoom text-center"
>
  <img
    alt={0}
    className="full-width animated zoomIn faster"
    src="https://images.dog.ceo/breeds/labrador/n02099712_1150.jpg"
    title={0}
  />
</div>
`;
